devdoc/CodeReviews.md

```diff
@@ -0,0 +1,55 @@
+- Contributors should feel free to push back if they consider comments are too picky.  The reviewer can either agree, or provide reasons why they consider it to be an issue.
+- Sometimes a significant design problem means the rest of the code has not been reviewed in detail.  If this is the case it should be explicitly noted.
+- Repeated issue.  The reviewer is free to comment on every instance of a repeated issue, but a simple annotation should alert contributor to address appropriately eg: [Please address all instances of this issue]
+- The contributor should respond to comment if it isn't obvious where/how they have been addressed (but no need to acknowledge typo/indentation/etc)
+- Only the reviewer should mark issues as resolved using the Github resolve conversation button.
+- Both reviewers and contributors should respond in a timely manner - don't leave it for days.  It destroys the flow of thought and conversation.
+- Watch out for collapsed conversations.  If there are large numbers of comments GitHib will collapse them which can make them easy to miss.
+- If there are large number of comments > 100, it may be better to close the PR and open a new one.
+- Submit any changes as extra commits so that it is clear to the reviewer what has changed, and avoids them having to re-review everything.  Please do not squash them until the reviewers approve the PR.  A few exceptions to this are if the only comments that the reviewer makes are indentation/typos, if the PR is only a couple of lines, or the PR is completely rewritten in response to the review.
+
+The "strictness" of the review should reflect the importance and location of the change.  Some examples:
+* If it is closely associated with an existing file, then the indentation, style should match the surrounding code - a mixture of styles makes it much harder to read.  If it is in a new, independent source file or project this is less of an issue.
+* If the code is in a core library then efficiency and edge cases will be more important.
+* If it is a core part of the system then security is key.  If it is a developer tool then edge cases are less significant.
+
+When reading comments in a review it can sometimes be hard to know why the reviewer made a comment, or what response is expected.  If there is any doubt the contributor should ask.  However to make it clearer we are aiming to always add a tag to the front of each review comment.  The tag will give an indication of why the comment is being made, its severity and what kind of response is expected.  Here is a provisional table of tags:
+
+Tag | What | Why | Expected response
+--- | ---- | --- | -----------------
+design: | An architectural or design issue | The reviewer considers the PR has a significant problem which will affect its functionality or future extensibility | reviewer/developer redesign expected before any further changes
```

Comment:
personal: covered by 'design' probably, but am wondering whether 'scope' is a suitable tag in itself. It is not uncommon for a PR to be too broad for one PR, where functionality has been added/combined in a single PR, that could have better been broken down into smaller more digestible chunks, or more appropriate for future work.

Reply:
added